### DISCUSSION TITLE
Citation inside of math code block

### DISCUSSION THREAD

**Discussion by mainrs** [typst/typst#4415] [Q&A]
I want to add a citation after a definition I introduce. My typst code looks like this:

```typst
Let $X$ be a random variable with a distribution $p$. The entropy of $X$ is defined as

$ HH(X) = -sum_(k=1)^K p(X = k) log_2(p(x = k)) @<key>. $
```

Note the dot inside the math block, since if I add it afterwards, it is added as a new symbol on the next line. I'd like the citation to be before the dot. However, using the `@<key>` format is a syntax error.

**Comment by l0wx** [comment]
From a stylistic POV, I'd advise against both, as inline `display()`  math will fringe your running text, but the latter at least isn't a complete hack and doesn't lead to (resp. requires mitigation of) mid-sentence block spacing.

```typst
#show math.equation.where(block: true): set align(left)

Let $X$ be a random variable with a distribution $p$. The entropy of $X$ is defined as

$ HH(X) = -sum_(k=1)^K p(X = k) log_2(p(x = k)) #text[@key.] $

---

Let $X$ be a random variable with a distribution $p$. The entropy of $X$ is defined as\
$display(HH(X) = -sum_(k=1)^K p(X = k) log_2(p(x = k)))$ @key.
```
